packages/headless/src/controllers/generated-answer/headless-generated-answer.test.ts

```diff
@@ -275,6 +275,34 @@ describe('generated answer', () => {
 
       expect(action).toBeTruthy();
     });
+
+    describe('when we have multiple controllers', () => {
+      let secondEngine: MockSearchEngine;
+      function callSecondListener() {
+        return (secondEngine.subscribe as jest.Mock).mock.calls.map(
+          (args) => args[0]
+        )[0]();
+      }
```

Comment:
To execute the second listener that was added we need to target the index `1` instead of `0` to get something like this:
```
      function callSecondListener() {
        return (secondEngine.subscribe as jest.Mock).mock.calls.map(
          (args) => args[0]
        )[1]();
      }
```
But for a more generic solution I would only update the `callListener` function and make it execute every listener that was added: 
```
    function callListener() {
      return (engine.subscribe as jest.Mock).mock.calls
        .map((args) => args[0])
        .forEach((listener) => {
          listener();
        });
    }
```

Reply:
nice catch! 👀 Thank youu